bin/upversion

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash -e
+#
+# Usage:
+#   bin/upversion NEW_VERSION
+# 
+# Changes the version where it is hardcoded in UDMI files, performing checks
+# that the new version is available, and that the files can be safely updated.
+# Modifications:
+#   - schema/*.json (all files)
+#   - etc/categories.json 
+#   - tests/**/*.json and validator/traces/**/*.out - 
+#     all files must be added to etc/upversion.txt only files preceeded
+#     with a 'y' are updated. Only works with "pretty"/non-minified JSON
+#   - documentation in-line exammples
+#   - specific java files (Pubber.java, ConfigUtil.java, LocalDevice.java)
+
+UPVERSION_LIST=etc/upversion.txt
+TESTS_REGEX="^(\s{0,4}\"version\"\s*:\s*)([0-9.\"]*)(,?)"
+SCHEMA_VERSION_IDENTIFIER=\$udmi_version # must start with $
+
+# Updates existing $udmi_version in a JSON file. Called by 
+# update_schema_version
+# Usage: update_existing_schema_version FILE_TO_UPDATE NEW_VERSION
+function update_existing_schema_version(){
+    file=$1
+    version=$2
+    gsed -i -E "s/\"\\$SCHEMA_VERSION_IDENTIFIER\"\s*:\s*\"[0-9.]*\"/\"$SCHEMA_VERSION_IDENTIFIER\": \"$version\"/" $file
+}
+
+# Updates an existing $udmi_version, or adds if missing
+# Usage: update_schema_version FILE_TO_UPDATE NEW_VERSIO
+function update_schema_version(){
+    file=$1
+    version=$2
+    if [[ $(jq -r ".[\"$SCHEMA_VERSION_IDENTIFIER\"]" $file) == "null" ]]; then
+        gsed -i  "s/^{/{\n  \"$SCHEMA_VERSION_IDENTIFIER\": \"$version\",/" $file
+    else
+        update_existing_schema_version $file $version
+    fi 
+}
+
+# Updates UDMI_VERSION variable in a java file 
+# Usage: update_java_file FILE_TO_UPDATE NEW_VERSION
+function update_java_file(){
+    file=$1
+    version=$2
+    [[ $(ggrep -c "UDMI_VERSION\s*=\s*\"" $file) != 1 ]] && return 1
+    gsed -i -e "s/UDMI_VERSION\s*=\s*\"[0-9.]*\"/UDMI_VERSION = \"$version\"/" $file
+}
+
+if [[ "$#" != 1 ]]; then
+    echo Usage: $0 NEW_VERSION
+    exit 1
+fi
+NEW_VERSION=$1
+shift 1
+
+if ! [[ $NEW_VERSION =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+    echo ERROR Invalid version: $NEW_VERSION
+    # exit 1
```

Comment:
Remove comments for dead code. In the long run "# exit 1" isn't useful for anything...

Reply:
all the `exit 1` shouldn't have been commented, they have been uncommented